it/src/test/java/com/sonar/runner/it/DistributionTest.java

```diff
@@ -0,0 +1,182 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonarqube.ws.WsMeasures.Measure;
+
+import static java.lang.Integer.parseInt;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DistributionTest extends ScannerTestCase {
+
+  private static final int SCRIPT_TIMEOUT_SECONDS = 15;
+
+  @ClassRule
+  public static TemporaryFolder temp = new TemporaryFolder();
+
+  private static Path sonarScannerPath;
+
+  enum OS {
+    LINUX,
+    WINDOWS,
+    MACOSX
+  }
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws IOException {
+    Path workDir = temp.newFolder().toPath();
+
+    String version = artifactVersion().toString();
+    OS os = getOS();
+    String zipPath = String.format("../target/sonar-scanner-%s-%s.zip", os.name().toLowerCase(), version);
+    File zipFile = new File(zipPath);
+    assertThat(zipFile).isFile();
+
+    unzip(zipFile, workDir);
+
+    Path scannerHome = Files.list(workDir).findFirst().get();
+    Path javaPath;
+
+    switch (os) {
+      case LINUX:
+      case MACOSX:
+        sonarScannerPath = scannerHome.resolve("bin/sonar-scanner");
+        sonarScannerPath.toFile().setExecutable(true);
+        javaPath = scannerHome.resolve("lib/jre/bin/java");
+        javaPath.toFile().setExecutable(true);
+        break;
+      case WINDOWS:
+        sonarScannerPath = scannerHome.resolve("bin/sonar-scanner.bat");
+        javaPath = scannerHome.resolve("lib/jre/bin/java.exe");
+        assertThat(javaPath).isRegularFile();
+        break;
+    }
+
+    assertThat(sonarScannerPath).isExecutable();
+  }
+
+  @After
+  public void cleanup() {
+    orchestrator.resetData();
+  }
+
+  @Test
+  public void script_should_push_report_to_sonarqube() throws IOException, InterruptedException {
+    String projectKey = "java:basedir-with-source";
+    orchestrator.getServer().restoreProfile(ResourceLocation.create("/sonar-way-profile.xml"));
+    orchestrator.getServer().provisionProject(projectKey, "Basedir with source");
+    orchestrator.getServer().associateProjectToQualityProfile(projectKey, "java", "sonar-way");
+
+    File projectDir = new File("projects/basedir-with-source");
+    executeShellScript(projectDir, projectKey, orchestrator);
+
+    Map<String, Measure> projectMeasures = getMeasures(projectKey, "files", "ncloc");
+    assertThat(parseInt(projectMeasures.get("files").getValue())).isEqualTo(1);
+    assertThat(parseInt(projectMeasures.get("ncloc").getValue())).isGreaterThan(1);
+  }
+
+  private void executeShellScript(File projectDir, String projectKey, Orchestrator orchestrator) throws IOException, InterruptedException {
```

Comment:
All this logic to install/run the scanner mostly duplicate what is in Orchestrator. Would be probably simpler to modify Orchestrator to be able to use a native scanner. Something like `SonarScanner.create().setScannerVersion(X.Y).useNative(true)`. In SonarScannerInstaller you would just have to append the correct classifier depending on the OS.